core-processor/src/common.rs

```diff
@@ -321,12 +318,18 @@ pub enum ExecutionErrorReason {
     /// Changed page has no data in initial pages
     #[display(fmt = "Changed page has no data in initial pages")]
     PageNoData,
-    /// Ext works with lazy pages, but lazy pages env is not enabled
-    #[display(fmt = "Ext works with lazy pages, but lazy pages env is not enabled")]
-    LazyPagesInconsistentState,
     /// Page with data is not allocated for program
     #[display(fmt = "{:?} is not allocated for program", _0)]
     PageIsNotAllocated(PageNumber),
+    /// Lazy pages init failed for current program.
+    #[display(fmt = "Cannot init lazy pages for program: {}", _0)]
+    LazyPagesInitFailed(String),
+    /// Cannot read initial memory data from wasm memory.
+    #[display(fmt = "Cannot read data for {:?}: {}", _0, _1)]
+    InitialMemoryReadFailed(PageNumber, MemoryError),
+    /// Cannot write initial data to wasm memory.
+    #[display(fmt = "Cannot write intial data for {:?}: {}", _0, _1)]
+    InitialDataWriteFailed(PageNumber, MemoryError),
```

Comment:
Can't we wrap these and similar variants with some general option of `ExecutionErrorReason`
Like `ExecutionErrorReason::Module( **these vars** )`

Reply:
I just want to reduce amount of variants in `ExecutionErrorReason`

Reply:
I think it's not bad to have a lot of error cases. May be you can produce propose which will be informative as this and contains less errors in enum?

Reply:
I'm talking about only this enum `ExecutionErrorReason`
I suggest to provide something like
```rust
enum MemoryStateError {
/// Cannot read initial memory data from wasm memory.
     #[display(fmt = "Cannot read data for {:?}: {}", _0, _1)]
     InitialMemoryReadFailed(PageNumber, MemoryError),
     /// Cannot write initial data to wasm memory.
     #[display(fmt = "Cannot write intial data for {:?}: {}", _0, _1)]
     InitialDataWriteFailed(PageNumber, MemoryError),
}
```
and reuse it in `ExecutionErrorReason`:

```rust
enum ExecutionErrorReason {
  // .. //
  MemoryState(MemoryStateError),
}
```

Reply:
And I totally agree with many error variants to make debugging and throwing errors as concrete and clear as it could be